simple_parsing/helpers/serialization/encoding.py

```diff
@@ -81,17 +98,16 @@ def encode(obj: Any) -> Union[Dict, List, int, str, bool, None]:
 @encode.register(tuple)
 # @encode.register(Sequence) # Would also encompass `str!`
 @encode.register(set)
-def encode_list(obj: Iterable) -> Sequence:
+def encode_list(obj: Union[List[Any], Set[Any], Tuple[Any, ...]]) -> List[Any]:
     # TODO: Here we basically say "Encode all these types as lists before serializing"
     # That's ok for JSON, but YAML can serialize stuff directly though.
     # TODO: Also, with this, we also need to convert back to the right type when
     # deserializing, which is totally doable for the fields of dataclasses,
     # but maybe not for other stuff.
     return list(map(encode, obj))
 
-@encode.register(dict)
 @encode.register(Mapping)
-def encode_dict(obj: Mapping) -> Dict:
+def encode_dict(obj: Mapping[Any, Any]) -> Dict[Any, Any]:
```

Comment:
this `Mapping` is from `collections.abc` but not `typing`. Therefore remove generics info